Add py35 py36-dev tests, remove failing 2.6 test, move regex compilation for nice speedup

ssteinerx opened the pull request:
As per PR title.  See commit msgs for a little more detail.

ssteinerx committed:
Add python 3.5 and 3.6-dev to tox tests

ssteinerx committed:
Remove already failing py26 test env

OrderdDict was causing doctest failure in released version anyway.

ssteinerx committed:
Move regex compilation out of parse() function

ssteinerX: ~10x speed improvement on execution of parse() function according to timing given by test runner.  Didn't benchmark further. 

Also simplified assignment of 'NaN' to all result values on exception, avoiding ['NaN']*4 operation for failed calls.

ssteinerx committed:
Update .gitignore to skip egg-info, setup.py use setuptools

ssteinerx commented:
Also updated .gitignore to skip .egg-info dir and switched to setuptools from distutils to allow use of:

    python setup.py develop

and all the other attendant improvements from setuptools.

ssteinerx committed:
Remove 2.6 from list of supported Python versions in setup.cfg

ssteinerx commented:
This passes tests but does not successfully parse the output from ping on my OS X system.

Will open new one when I've fixed whatever is causing that issue, too.

gg commented:
Thanks for your contributions. Sorry, I haven't been able to test this on OS X. I wrote this as a quick-and-dirty way to parse ping results for myself on Linux and familiarize myself with Python packaging.

I should update the README to direct people to the [pingparsing library](https://github.com/thombashi/pingparsing) as it's more comprehensive and better maintained than my library. If you're looking to contribute, I think it's more worthwhile to test/add OS X compatibility to the pingparsing library.

ssteinerx committed:
Added test data and tests to tox for all failing variants

ssteinerx commented:
I just whacked the heck out of it.  There is now a place for test data i.e. the output of the ping command with the expected result as the first line and it's in tox.

I'll go look at pingparsing and add my tests there as well.

ssteinerx commented:
Doesn't look like they tested OS X either.

ssteinerx commented:
No worries.  

I have need for a ping in a product I'm writing and had to have thorough tests to which I can add data from any systems that cause exceptions anyway so no wasted effort.

ssteinerx commented:
Just took a look at pingparsing.

It's not more comprehensive, it's just more complicated and unnecessarily so.  I find the code very un-Pythonic and I don't like pyparsing over regular expressions for something this simple.

If I have to extend one, I'll extend this rather than work on that.

gg commented on src/pingparser.py:
```diff
@@ -73,39 +126,42 @@ def main(argv=sys.argv):
-    for (fmt, rep) in format_replacements:
-        output = output.replace(fmt, rep)
-
+    format_string = default_format
```
Custom formatting is not working. I think this should be:

```python
    format_string = options.format or default_format
```

ssteinerx replied:
Done.

gg reviewed (changes requested):
Thanks, I appreciate the improvements! I'm ready to merge this in after you commit a quick fix for custom formatting.

gg commented:
> It's not more comprehensive, it's just more complicated and unnecessarily so. I find the code very un-Pythonic and I don't like pyparsing over regular expressions for something this simple.

Ah, I didn't bother to dig into the code. After a cursory glace, it appeared to have more features and better documentation/tests. Like you said, no worries!

ssteinerx commented:
I may have made a mistake there -- there isn't a set of tests for that and I'm interested in using this as a library so my focus wasn't on the command line utility, but on the backend functionality.

You could practice and write a test, or I can get to it maybe tomorrow -- right now I'm in the middle of a live site backend credit card processing nightmare on a site whose busiest days of the year are this week (of course).

S

On Nov 21, 2016, at 6:40 PM, Gregg Gajic wrote:

> @gg requested changes on this pull request.
> 
> Thanks, I appreciate the improvements! I'm ready to merge this in after you commit a quick fix for custom formatting.
> 
> In src/pingparser.py:
> 
> > -                           ('%j', 'jitter')]
> -    format_replacements = [(fmt, ping_result[field]) for fmt, field in
> -                           format_replacements]
> -
> -    if len(args) == 0:
> -        output = ','.join(fmt for (fmt, rep) in format_replacements)
> -    elif args[0].startswith('+'):
> -        args[0] = args[0].lstrip('+')
> -        output = ' '.join(args[0:])
> -    else:
> -        parser.print_help()
> -
> -    for (fmt, rep) in format_replacements:
> -        output = output.replace(fmt, rep)
> -
> +    format_string = default_format
> ATM custom formatting is not working. I think this should be:
> 
>     format_string = options.format or default_format
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

ssteinerx committed:
Update .gitignore to skip build/ & dist/, move pingparser.py to root

Apply options.format if passed, note about old format overriding new
(maybe should be warning)

test.py is now unittest.TestCase based

Sketched in command line testing, not implemented.  See note.

ssteinerx committed:
Update command line to use -f instead of + syntax for custom format string

gg commented:
Thanks again!

ssteinerx commented:
That'd probably be enough for a version bump and release, yes?

gg commented:
> That'd probably be enough for a version bump and release, yes?

I bumped the version to 0.4 in setup.cfg.

As far as a release goes... Do you mind if I transfer repository ownership to you? Admittedly, I haven't had use for this project in *years*. I think it's in better off in your hands since you're using it and may make additional contributions in the future.

ssteinerx commented:
Sure, pass it on (though I'm not sure how to do that).

ssteinerx commented:
I'm not sure how to xfer the PyPi listing either...

gg commented:
> Sure, pass it on (though I'm not sure how to do that).

Before I can transfer the repo, you need to rename or remove [your fork of `pingparser`](https://github.com/ssteinerx/pingparser) since it has the same name as this repository.

> I'm not sure how to xfer the PyPi listing either...

AFAIK, pingparser isn't listed on PyPi.

ssteinerx commented:
My fork is renamed.

gg commented:
Sorry, it looks like you have to remove the fork. I'm getting this error message:

> ssteinerx already has a repository in the gg/pingparser network

For reference, I found a help page on repository transfers: https://help.github.com/articles/about-repository-transfers/

ssteinerx commented:
Gone!

gg commented:
Transfer request sent.
